devices/devs.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Phoenix-RTOS
+ *
+ * phoenix-rtos-loader
+ *
+ * Device Interface
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "devs.h"
+#include "../errors.h"
+
+#define DEV_MAX_NB            8
+#define DEV_MAX_INSTANCE_NB   16
+
+/* dev_driver_t go to .bss section where is not zeroed
+ * This magic number defines whether driver was registered */
+#define DRIVER_IS_ACTIVE      0xababcdcd
+
+typedef struct {
+	int (*init)(int dn, dev_handler_t *h);
+	int active;
+} dev_driver_t;
+
+
+struct {
+	dev_handler_t devs[DEV_MAX_NB][DEV_MAX_INSTANCE_NB];
```

Comment:
not sure if we want to have two-dimension array here, seems like a waste of space (I don't know about the use cases and if it's really important)

Reply:
The two-dimension array was introduced here to keep handlers of instances of all devices. For example, imxrt107x has 12 UART instances so that we need to be ready to keep them all.

Reply:
yeah, I understand WHY we need it (it's major/minor), just usually we will have 12 uarts and single ethernet / flash/ etc, so for most devices the other instances would be empty. I like the simplicity so If we're not afraid of the memory usage it's ok. We can always change the implementation to hashtable/linked list if that would provide to be problem on smaller targets.